sycl/include/sycl/platform.hpp

```diff
@@ -57,6 +62,40 @@ namespace ext::oneapi {
 class filter_selector;
 } // namespace ext::oneapi
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+template <typename ParamT>
+typename detail::ABINeutralT_t<ParamT> convert_to_abi_neutral(ParamT &Info) {
```

Comment:
This helper itself should be in `sycl::detail`.

Reply:
They are already defined in sycl::detail namespace. 
Scroll up a few more lines, you will see `namespace detail {` at line 48.
Are you referring to something else?

Reply:
The one that ends at line 59 and forces you to use `detail::ABINeutralT_t` instead of just `ABINeutralT_t`?

Reply:
I missed that line. Thanks for pointing it out.

Reply:
It seems I can't avoid the `detail::` part even if I moved these helper function into the `sycl::_V1::detail` namespace.
```
/localdisk2/bso/sycl/llvm-pr/build/include/sycl/platform.hpp:62:10: error: expected nested-name-specifier
   62 | typename ABINeutralT_t<ParamT> convert_to_abi_neutral(ParamT &Info) {
      |          ^~~~~~~~~~~~~~~~~~~~~
/localdisk2/bso/sycl/llvm-pr/build/include/sycl/platform.hpp:62:32: error: invalid declarator before ‘convert_to_abi_neutral’
   62 | typename ABINeutralT_t<ParamT> convert_to_abi_neutral(ParamT &Info) {
      |                                ^~~~~~~~~~~~~~~~~~~~~~
```
Then, also all the calls to these functions should be prefixed with `detaill:`: to call from within the class device and platform.

Reply:
> It seems I can't avoid the detail:: part even if I moved these helper function into the sycl::_V1::detail namespace

That doesn't make sense, I'd bet you have some other issue that needs to be addressed too. My first thought would be to drop `typename` - it's not needed for an alias template.

> Then, also all the calls to these functions should be prefixed with detaill:: to call from within the class device and platform

That's not different from any other helper anywhere in the project.

Reply:
Excellent point.
I removed typename and detail:: together, and now it works.
Thanks a lot, Andrei.